### DISCUSSION TITLE
Create models based on MetricFlow queries

### DISCUSSION THREAD

**Discussion by michelbouchou** [dbt-labs/dbt-core#11119] [Ideas]
### Current behavior

I am using MetricFlow with dbt-core. 
dbt-core is not meant to execute saved-queries based on created metrics but only allow to generate SQL. Only dbt Cloud handles the export functionality.

### Idea

I suggest to add a macro that can compile SQL based on a saved-query and handle it in a new model. It would allow dbt-core users to have a better experience using MetricFlow by handling the execution and new transformation after the metric definition part.  
Some user published a similar macro on Github a couple of months ago but this is not working correctly and won't be maintained as part of dbt and MetricFlow :
https://github.com/DJLemkes/dbt-metric-utils

**Comment by 1cadumagalhaes** [comment]
Why is this a discussion and not a feature request? This sounds very good. It would make way more sense to be able to use metrics inside of models, they would be like a macro.
